### ISSUE TITLE
Editor re-write design discussion

### ISSUE THREAD

**Issue by prock-fife** [fifengine/fifengine-editor#29]
I think it's finally time someone tackles the editor.  It's current state is basically unusable.  I'd like to collect a list of features and prioritize them here.  **NOTE** this list will grow as we discuss.. it is a work in progress.

Must have features - **High Priority** (Let's not talk how we will implement this stuff yet):
- Load/save map
- Edit global map settings
- Change view (rotate, zoom, pan, etc etc)
- Re-arrange layers (re-order)
- Select layer
- Show/Hide layer
- Add/Edit/Remove layer
- Select object to "draw"/place
- Add (place)/Edit/Remove Instance
- Add/Edit/Remove Objects (including actions)
- Edit Cell properties (includes pathfinding properties)
- Toggle map/layer grid
- Toggle layer cell coordinates

**Medium Priority**
- Add/Edit/Remove additional cameras
- Add/Edit/view pathfinder transition between layers
- Add/Edit/Remove Lights
- Add/Edit/Remove Triggers
- Add/Edit/Remove Custom attributes (need to add a callback for custom attributes to FIFE loaders maybe?)
- Add/Edit/Remove Sound Sources (global/ambient sounds)

**Low Priority**
- Tileset Editor/Creator

**Behaviors**
- All object references should be relative to the map file location

**Ideas**
- Add ability to specify "include" directories for object references
- Editor meta file or "project" file to store project specific configurations
  - Default camera settings
  - Default "include" directories
- Ability to "install" the map into the correct location for client testing.  This might include the ability to bundle the map and resources.

**Comment by LinuxDonald** [comment]
The fiferpg editor (done in cegui) now works too with fife maps: https://github.com/fife-rpg/editor/commit/3aae8f6ea6b2cdd3a13fa10f4cfb8208316af647

https://github.com/fife-rpg/editor

**Comment by helios2000** [comment]
Really cool! Good work @Beliaar 👍

**Comment by LinuxDonald** [comment]
What about an merge of fife-rpg-editor into this repo here? @helios2000 @prock-fife @vtchill

**Comment by Beliar83** [comment]
Just a note, i only added support for loading maps without a fife-rpg project, but i did not remove any fife-rpg specific features.

**Comment by LinuxDonald** [comment]
I don't think we would merge it now. But as an long term thing. We could do it instand of write an complete new one here.
